src/util/adb.js

```diff
@@ -210,10 +210,45 @@ export default class ADBUtils {
     await this.runShellCommand(deviceId, ['mkdir', '-p', artifactsDir]);
 
     this.artifactsDirMap.set(deviceId, artifactsDir);
-
     return artifactsDir;
   }
 
+  async checkOrCleanArtifacts(
+    deviceId: string, remove?: boolean): Promise<boolean> {
+    const {adbClient} = this;
+
+    let directories = false;
+
+    await wrapADBCall(async () => {
+      log.debug('Finding older artifacts');
+      const files = await adbClient.readdir(deviceId, '/sdcard');
+
+      for (const file of files) {
+        if (file.isDirectory() &&
+              file.name.startsWith('web-ext-artifacts-')) {
+          directories = true;
+
+          if (!remove) {
+            break;
+          }
```

Comment:
we can avoid this nested if/else blocks, in my previous review comment there should be some pointers about how to achieve that.

Reply:
In the meta code that you provided,
for (let file of filenames) {
  if (file is not a directory or
      its name doesn't start with the artifacts name prefix) {
     continue; // no need to do anything else on this item, move to the next
  }

  hasOldArtifacts = true;

  if (this should delete the old artifacts) {
     remove the artifact dir
  }
}

Have the same level of nested if/else blocks and in this as you pointed that if we have to only check the condition for old artifacts then if one is found we should break out of the loop, which wasn't possible in the meta code